vanguard.go

```diff
@@ -393,6 +393,21 @@ func WithMaxGetURLBytes(limit uint32) ServiceOption {
 	})
 }
 
+// WithRESTUnmarshalOptions returns a service option that sets the unmarshal options for use with the REST protocol.
+func WithRESTUnmarshalOptions(options RESTUnmarshalOptions) ServiceOption {
+	return serviceOptionFunc(func(opts *serviceOptions) {
+		opts.restUnmarshalOptions = options
+	})
+}
+
+// RESTUnmarshalOptions contains options for unmarshalling REST requests.
+//
+// If DiscardUnknownQueryParams is true, any query parameters in a request that do not correspond to a field in the
+// request message will be ignored. If false, such query parameters will cause an error. Defaults to false.
```

Comment:
This should be moved down below, so it describes the field itself, instead of being in docs for the struct.

Reply:
Moved it down.